src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -36,6 +36,12 @@
         entity-datums (map (partial str "this.") (get-in entity-types [(keyword selected-entity-type) :attributeNames]))]
     (concat entity-datums workspace-datums)))
 
+(defn- fix-validated-method-config [config]
+  (let
+    [munged-missing (map (fn [missing-input] {(keyword missing-input) "This input is declared in the method and missing in your method configuration."}) (:missingInputs config))
+     munged-extras (map (fn [extra-input] {(keyword extra-input) "This input is declared in your method configuration but not listed in the method."}) (:extraInputs config))
+     new-inputs (into (:invalidInputs config) (concat munged-missing munged-extras))]
+    (assoc config :invalidInputs new-inputs)))
```

Comment:
Being fairly lazy here, and folding the missing and extras lists into the list of invalid inputs with default messages, and pretends they were there all along. That way I don't have to update the MC render code.

Reply:
Seems like a reasonable approach.

You can use `update` to modify a value in a map, instead of manually extracting->updating->reinserting. Something like:
```clojure
(update config :invalidInputs into munged-missing munged-extras)
```

Which means `munged-missing` and `munged-extras` can be left as lists-of-pairs instead of maps:
```clojure
(let [munged-missing (map (fn [missing-input] [(keyword missing-input) "text"]) (:missingInputs config))
      ...])
```

And then for advanced fun,
```clojure
(fn [missing-input] [(keyword missing-input) "text"])
```
can be
```clojure
(juxt keyword (constantly "text"))
```

Reply:
This is more philosophy, but I think `(juxt keyword (constantly "text"))` would be a little confusing to people who aren't ui grown-ups.

Reply:
> `(update config :invalidInputs into munged-missing munged-extras)`

I think one of my in-between attempts looked like:

```
(update config :invalidInputs munged-missing)
```

which only returned the new value of `invalidInputs`, not the new value of `config`. I'll doublecheck this works, but for my understanding: this works because `into` plays the role of `f` [here](https://clojuredocs.org/clojure.core/update), right?

Reply:
yes